crates/uv-requirements/src/upgrade.rs

```diff
@@ -64,10 +64,13 @@ pub async fn read_requirements_txt(
 }
 
 /// Load the preferred requirements from an existing lockfile, applying the upgrade strategy.
-pub async fn read_lockfile(workspace: &Workspace, upgrade: &Upgrade) -> Result<LockedRequirements> {
+pub async fn read_lockfile(
+    workspace: &Workspace,
+    upgrade: &Upgrade,
+) -> Result<(Option<Lock>, LockedRequirements)> {
```

Comment:
Maybe we split this into two functions, one that reads the lockfile, and one that takes `Lock` and returns `LockedRequirements`? Seems more flexible to compose that way.